container/activities/src/test/java/org/mobicents/slee/runtime/activity/ActivityContextImplTest.java

```diff
@@ -0,0 +1,216 @@
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.mobicents.slee.runtime.activity;
+
+import junit.framework.Assert;
+import junit.framework.TestCase;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.management.Notification;
+import javax.slee.Address;
+import javax.slee.EventTypeID;
+import javax.slee.SLEEException;
+import javax.slee.ServiceID;
+import javax.slee.facilities.TimerID;
+import javax.slee.facilities.TraceLevel;
+import javax.slee.management.NotificationSource;
+import javax.slee.management.TraceNotification;
+import javax.slee.resource.ActivityFlags;
+import javax.slee.resource.ActivityHandle;
+import javax.slee.resource.ActivityIsEndingException;
+
+import org.apache.log4j.Logger;
+import org.infinispan.tree.Node;
+import org.mobicents.slee.container.activity.ActivityContext;
+import org.mobicents.slee.container.activity.ActivityContextCacheDataInterface;
+import org.mobicents.slee.container.activity.ActivityContextHandle;
+import org.mobicents.slee.container.activity.ActivityType;
+import org.mobicents.slee.container.management.jmx.TraceMBeanImpl;
+import org.mobicents.slee.container.sbbentity.SbbEntityID;
+import org.restcomm.cache.MobicentsCache;
+import org.restcomm.cluster.DataRemovalListener;
+import org.restcomm.cluster.DefaultMobicentsCluster;
+import org.restcomm.cluster.FailOverListener;
+import org.restcomm.cluster.MobicentsCluster;
+import org.restcomm.cluster.cache.ClusteredCacheDataIndexingHandler;
+import org.restcomm.cluster.election.DefaultClusterElector;
+
+
+public class ActivityContextImplTest extends TestCase {
+
+  public void testGetActivityFlagsNotInitialized() {
+	  ActivityContextHandle ach = new FakeActivityContextHandle();
+	  FakeActivityContextCacheData activityContextCacheData = new FakeActivityContextCacheData();
+	  ActivityContextImpl ac = new ActivityContextImpl(ach,activityContextCacheData, false, null);
+	  Assert.assertEquals(ac.getActivityFlags(), ActivityFlags.NO_FLAGS);

```

Comment:
Thanks @jaimecasero  for valuable comments :). I will change to use  mockito to create some mock function. Also Keep new created ActivityContextCacheDataInterface . does it sound ok ?